### PR TITLE
Improve API Platform tutorial with Overlays usage

### PR DESCRIPTION
**Pull request by ChristopheDujarric**

### CODE REVIEW

File: src/_guides/bump-sh-tutorials/api-platform.md [hunk 1]
```diff
@@ -13,17 +17,45 @@ The following assumes your local machine is configured with PHP and API Platform
...
     --token my-documentation-token
   ```
 
 That's it! Enjoy the comfort of Bump.sh to browse through your API doc, and [customize it to your needs](/help/getting-started/quick-start#customization-options).
+
+## Enriching definition files
+
+By default, API Platform generates what can be considered a plain technical description of your API. Although it is robust and complete, it's not enough for people discovering your API without any upfront knowledge about it, access to the code, or ability to test it extensively. Such definition files require contextual information, and any appropriate adjustements.
+
+API Platform lets you override, enrich and filter the generated OpenAPI definition files, [using PHP](https://api-platform.com/docs/core/openapi/).
```
Comment (philsturgeon) [inline src/_guides/bump-sh-tutorials/api-platform.md:39]: I've been trying to get folks to standardize on "OpenAPI description documents" as per this glossary, got a lot of various vendors agreeing and I've been following that for articles and guides written for you. https://github.com/openapi-contrib/glossary

File: src/_guides/bump-sh-tutorials/api-platform.md [hunk 2]
```diff
@@ -13,17 +17,45 @@ The following assumes your local machine is configured with PHP and API Platform
...
+
+You can also accomplish this using OpenAPI's [Overlay specification](https://github.com/OAI/Overlay-Specification/blob/main/versions/1.0.0.md), which can be convenient in all use cases where you do not want to change your PHP code base (e.g. when involving Technical Writers or Product Managers in writing docs).
+
+The Bump.sh CLI supports Overlays. Technically, the `bump overlay` command will output a modified version of the `[DEFINITION_FILE]` (an OpenAPI or AsyncAPI document) by applying the operations described in the `[OVERLAY_FILE]` Overlay file to the original API document.
+
+1. Create your Overlay document (see example use cases in [Augmenting Generated OpenAPI Documents with Filters & Overlays](/guides/openapi/augmenting-generated-openapi))
+
+2. Apply your Overlay document to the original API document with the Bump.sh CLI
+```shell
+bump overlay docs.json overlay-file.json > api-overlayed-document.json
```
Comment (philsturgeon) [inline src/_guides/bump-sh-tutorials/api-platform.md:49]: I'd stick to YAML and use standard names similar to the guide we wrote: 

```suggestion
bump overlay openapi.yaml overlays.yaml > openapi.updated.yaml
```

File: src/_guides/bump-sh-tutorials/api-platform.md [hunk 3]
```diff
@@ -13,17 +17,45 @@ The following assumes your local machine is configured with PHP and API Platform
...
+```
+
+3. Deploy the resulting file to Bump.sh
+  ```shell
+  bump deploy api-overlayed-document.json \
```
Comment (philsturgeon) [inline src/_guides/bump-sh-tutorials/api-platform.md:54]:
```suggestion
  bump deploy openapi.updated.yaml \
```

**Review by philsturgeon — COMMENTED** [review]
Looks good!

File: src/_guides/bump-sh-tutorials/api-platform.md (hunk 2 above)
Comment (ChristopheDujarric) [reply src/_guides/bump-sh-tutorials/api-platform.md:49]: I opted for JSON and the initial `docs.json` file name as it's the file name and format that is automatically generated by API Platform.

I wanted to make API Platform user feel more "at home", WDYT?

File: src/_guides/bump-sh-tutorials/api-platform.md (hunk 1 above)
Comment (ChristopheDujarric) [reply src/_guides/bump-sh-tutorials/api-platform.md:39]: Thanks for that feedback! I realize that I'm actually not entirely in sync with our own glossary https://docs.bump.sh/help/getting-started/concepts/#api-definition, where I should end up talking about an "OpenAPI document".

It's getting closer to what you mention. Not sure if you suggest we should go one step further, but it's a wider review than just this PR then 😄

File: src/_guides/bump-sh-tutorials/api-platform.md (hunk 2 above)
Comment (philsturgeon) [reply src/_guides/bump-sh-tutorials/api-platform.md:49]: Good point! Yeah best go with whatever their defaults are, unless you'd like to help them have a nicer experience. 

Or you can update the guide to help them make more standard output:

image


Either way, the `overlays.yaml` name and `openapi.something.yaml` is up to us so could be whatever.

File: src/_guides/bump-sh-tutorials/api-platform.md (hunk 1 above)
Comment (philsturgeon) [reply src/_guides/bump-sh-tutorials/api-platform.md:39]: ha so long as you're consistent with your own glossary then do whatever in this blog post, can worry about aligning the rest later.

File: src/_guides/bump-sh-tutorials/api-platform.md [hunk 4]
```diff
@@ -13,17 +17,40 @@ The following assumes your local machine is configured with PHP and API Platform
...
+
+API Platform lets you override, enrich and filter the generated OpenAPI documents, [using PHP](https://api-platform.com/docs/core/openapi/). 
+
+You can also accomplish this using OpenAPI's [Overlay specification](https://github.com/OAI/Overlay-Specification/blob/main/versions/1.0.0.md), which can be convenient in all use cases where you do not want to change your PHP code base (e.g. when involving Technical Writers or Product Managers in writing docs).
+
+The Bump.sh CLI supports Overlays. Technically, the `bump overlay` command will output a modified version of the `[DEFINITION_FILE]` (an OpenAPI or AsyncAPI document) by applying the operations described in the `[OVERLAY_FILE]` Overlay file to the original API document.
+
+1. Create your Overlay document (see example use cases in [Augmenting Generated OpenAPI Documents with Filters & Overlays](/guides/openapi/augmenting-generated-openapi)), and name it `overlays.json` (YAML and JSON are both supported and input and outputs of the CLI).
```
Comment (paulRbr) [inline src/_guides/bump-sh-tutorials/api-platform.md:45]:
```suggestion
1. Create your Overlay document (see example use cases in [Augmenting Generated OpenAPI Documents with Filters & Overlays](/guides/openapi/augmenting-generated-openapi)), and name it `overlays.json` (YAML and JSON are both supported as inputs for the CLI).
```

**Review by paulRbr — APPROVED** [review]
:rocket: Thanks for improving the guides :bow:

**Pull request merged**